java/client/src/test/java/glide/api/ByteArrayArgumentMatcher.java

```diff
@@ -0,0 +1,41 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api;
+
+import java.util.List;
+import org.mockito.ArgumentMatcher;
+
+/**
+ * Argument matcher for comparing lists of byte arrays.
+ *
+ * <p>It is used in Mockito verifications to assert that a method call was made with a specific list
+ * of byte arrays as arguments.
+ */
+public class ByteArrayArgumentMatcher implements ArgumentMatcher<List<byte[]>> {
+    List<byte[]> arguments;
+
+    /**
+     * Constructs a new ByteArrayArgumentMatcher with the provided list of byte arrays.
+     *
+     * @param arguments The list of byte arrays to match against.
+     */
+    public ByteArrayArgumentMatcher(List<byte[]> arguments) {
+        this.arguments = arguments;
+    }
+
+    /**
+     * Matches the provided list of byte arrays against the stored arguments.
+     *
+     * @param t The list of byte arrays to match
+     * @return boolean - true if the provided list of byte arrays matches the stored arguments, false
+     *     Sotherwise.
+     */
+    @Override
+    public boolean matches(List<byte[]> t) {
+        int length = t.size();
+
+        for (int index = 0; index < length; index++) {
```

Comment:
Oh
That worked just because `t` is the same object as `arguments`
Don't create `String`s from `byte[]`, don't compare arrays with `equals`

I think `Arrays.deepEquals` can do all business for you, just convert `List<byte[]>` to `byte[][]`

Reply:
BTW I want to have `CommandManager` fixed to use `byte[][]` insteadl of `List<byte[]>` for consistency with other existing API